transformation-scripts/semopenalex-domains.py

```diff
@@ -0,0 +1,306 @@
+trig_output_dir_path = os.path.join(absolute_path, '../graphdb-preload/graphdb-import/') 
+trig_output_file_path = f'{trig_output_dir_path}{ENTITY_TYPE}-semopenalex-{today}.trig'
+
+data_dump_start_time = time.ctime()
+print('domains entity files started to download at: '+ data_dump_start_time)
+# Copy domains entity snapshot
+client = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+file_names, folders = get_file_folders(client, "openalex", "data/domains/")
+download_files(client, "openalex", data_dump_input_root_dir, file_names, folders)
+print('domains entity files finished to download.')
+
+start_time = time.ctime()
+print('domains entity started to transform at: '+ start_time)
+
+with open(trig_output_file_path, "w", encoding="utf-8") as g:
+
+    #Path where the OpenAlex data for the current entity type is located
+    data_dump_input_entity_dir = f'{data_dump_input_root_dir}/data/{ENTITY_TYPE}/*'
+
+    # initialize and add domain scheme URI
+    domain_graph.add((topic_scheme_uri, RDF.type, SKOS.ConceptScheme))
+    domain_graph.add((topic_scheme_uri, SKOS.prefLabel, Literal("SemOpenAlex Topics", datatype = XSD.string)))
+    domain_graph.add((topic_scheme_uri, URIRef("http://purl.org/dc/terms/description"), Literal("SemOpenAlex topics are abstract ideas that works are about. Topics are grouped into subfields, which are grouped into fields, which are grouped into top-level domains.", datatype = XSD.string)))
+
+    for filename in glob.glob(os.path.join(data_dump_input_entity_dir, '*.gz')):
+        with gzip.open(filename, 'r') as f:
+
+            for line in f:
+                try:
+                    json_data = json.loads(line.decode('utf-8'))
+
+                    #domain-ID
+                    domain_id = json_data['id'].replace("https://openalex.org/domains/", "")
+                    domain_uri = URIRef(soa_namespace_domains+domain_id)
+                    domain_graph.add((domain_uri,RDF.type, soa_class_domain))
+                    domain_graph.add((domain_uri,RDF.type, SKOS.Concept))
+
+                    #topic Scheme
+                    domain_graph.add((domain_uri,SKOS.inScheme,topic_scheme_uri))
+
+                    # display_name
+                    domain_display_name = json_data['display_name']
+                    if not domain_display_name is None:
+                        domain_display_name = clean(domain_display_name)
+                        domain_graph.add((domain_uri, SKOS.prefLabel, Literal(domain_display_name, datatype=XSD.string)))
+
+                    # display_name_alternatives
+                    domain_display_name_alternatives = json_data['display_name_alternatives']
+                    if not domain_display_name_alternatives is None:
+                        domain_display_name_alternatives = clean(domain_display_name_alternatives)
+                        domain_graph.add((domain_uri, SKOS.altLabel, Literal(domain_display_name_alternatives, datatype=XSD.string)))
```

Comment:
`display_name_alternatives` is array. you will need to loop.

Reply:
Done.